modal/cli/run.py

```diff
@@ -384,12 +384,18 @@ def shell(
     modal shell
     ```
 
-    Start a bash shell using the spec for `my_function` in your app:
+    Start a bash shell using the spec for `my_function` in your App:
```

Comment:
Unrelated, but we don't really explain what a "spec" is anywhere (don't even know if we have an official name for "the group of options that you can set on a function that affect what the container environment looks like") - cc @mwaskom since this ties back to the decorator brainstorming you've been doing

Reply:
Yeah agreed — was talking to another Modal eng recently who was confused about the difference between `Function` / `api_pb2.Function` / `FunctionInfo` / `FunctionSpec`. I don't believe that we define the "spec" externally anywhere so it's also not clear to users what parameters will be picked up by `modal shell` (@thundergolfer this is part of why I'm a little resistant to exposing this feature in other CLI tools).

If we embrace additional decorators it would be nice to be able to define the "spec" as "everything in `app.function(...)` and not need reference to a separate concept, but I suspect that won't be possible. First because I think autoscaling parameters probably still belong in `@app.function` rather than a separate decorator (but open to arguments from converts who are even more zealous than me... 😄) and also because we have some random parameters that don't really correspond to the "container spec" but that also don't really seem to merit their own decorators (`serialized`, `is_generator`, `enable_memory_snapshot`, etc.).

Reply:
> this is part of why I'm a little resistant to exposing this feature in other CLI tools

Yes I noticed that usage of "spec" as potentially something we're too vague on. I think of it as '`modal.Function` but with the body code replaced with a shell', but that's actually wrong on the details you write out below.

Reply:
Yeah, in fact it's probably surprising from a certain perspective that you can define a function with `concurrency_limit=1` and then "connect two shells" to it at the same time. Which isn't exactly that we're doing, but it's what it _looks_ like we're doing.